creusot/src/util.rs

```diff
@@ -248,31 +248,75 @@ pub(crate) fn module_name(tcx: TyCtxt, def_id: DefId) -> Symbol {
     }
 }
 
-pub(crate) fn ident_path(tcx: TyCtxt, def_id: DefId) -> Symbol {
-    use heck::ToUpperCamelCase;
+// Translate a name to be a valid fragment of a Why3 identifier
+// Escape initial and final underscores, double underscores, non-ascii characters,
+// and "qy" sequences (because "qy" is the escape sequence).
+// "qy123z" encodes the code point 123.
+fn push_translate_name(n: String, dest: &mut String) -> (){
+    let mut chars = n.chars().peekable();
+    // Escape initial underscore
+    if chars.peek() == Some(&'_') {
+        let _ = chars.next();
+        dest.push('q');
+        dest.push('y');
+        dest.push('9');
+        dest.push('5');
+        dest.push('z');
```

Comment:
You should be able to use `.extend("qy95z")` here.

Reply:
I found `.push_str()` which is similar. Do you have any preference vs `.extend()`?

Reply:
thats good. I forgot that `String` has a weird api there for whatever reason.